src/arim/geometry.py

```diff
@@ -1457,11 +1459,144 @@ def points_in_rectbox(
+            raise ValueError("Too many / few values of `numpoints` provided.")
+        
+    if names is not None:
+        if len(names) != coords.shape[0] - 1:
+            raise ValueError("Too many / few wall names provided.")
+    else:
+        bw_idx, sw_idx, ow_idx = 0, 0, 0
+    
+    walls = []
+    for idx, (start, end) in enumerate(zip(coords[:-1, :], coords[1:, :])):
+        if numpoints.shape[0] == 1:
+            n = numpoints[0]
+        else:
+            n = numpoints[idx]
+        
+        if names is None:
+            # Frontwall is first and has z == 0.0
+            if idx == 0 and abs(start[-1]) < np.finfo(float).eps and abs(end[-1]) < np.finfo(float).eps:
+                name = "frontwall"
+            # Backwall has constant z.
+            elif abs(start[-1] - end[-1]) < np.finfo(float).eps:
+                name = "backwall_{}".format(bw_idx)
+                bw_idx += 1
+            # Sidewall has constant x.
+            elif abs(start[0] - end[0]) < np.finfo(float).eps:
+                name = "sidewall_{}".format(sw_idx)
+                sw_idx += 1
+            else:
+                name = "otherwall_{}".format(ow_idx)
+                ow_idx += 1
+        else:
+            name = names[idx]
+        
+        walls.append(points_1d_wall(start, end, n, name=name, dtype=dtype))
+        
+    return walls
+
+
+def points_1d_wall(start, end, numpoints, name=None, dtype=None):
+    """
+    Returns a set of regularly spaced points between `start` and `end`.
+    
+    Orientation will always have x_hat in the direction of wall start -> end,
+    y_hat = j and z_hat = x_hat ^ j.
+
+    Parameters
+    ----------
+    start : ndarray[float]
+        1D array of length 2 or 3.
+    end : ndarray[float]
+        1D array of length 2 or 3.
+    numpoints : int
+    name : str or None, optional
+    dtype : type or None, optional
+
+    Returns
+    -------
+    OrientedPoints.
+
+    """        
+    if dtype is None:
+        dtype = s.FLOAT
+    start, end = np.squeeze(start), np.squeeze(end)
+    if start.shape[0] == 2:
+        start = np.asarray([start[0], 0.0, start[1]])
+    if end.shape[0] == 2:
+        end = np.asarray([end[0], 0.0, end[1]])
+    
+    # Make points and orientations
+    points = Points.from_xyz(
+        x = np.linspace(start[0], end[0], numpoints, dtype=dtype),
+        y = np.linspace(start[1], end[1], numpoints, dtype=dtype),
+        z = np.linspace(start[2], end[2], numpoints, dtype=dtype),
+        name=name,
+    )
+    
+    basis = CoordinateSystem(
+        [0.0, 0.0, 0.0],
+        (end - start) / np.linalg.norm(end - start), 
+        [0.0, 1.0, 0.0],
+    ).basis_matrix
+    
+    orientations_arr = np.broadcast_to(
+        basis, (*points.shape, 3, 3)
+    )
+    orientations = Points(orientations_arr)
+    
+    return OrientedPoints(points, orientations)
+
+
+def points_1d_wall_z(xmin, xmax, z, numpoints, y=0.0, name=None, is_block_above=True, dtype=None):
```

Comment:
In a multi-layer block, how would you define `is_block_above`?

Reply:
I haven't really considered multi-layer blocks, but I suppose it would come from what the orientation should be - i.e. z_hat pointing upwards or downwards.